aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -339,6 +344,8 @@ public AwsProperties(Map<String, String> properties) {
 
     this.dynamoDbTableName = PropertyUtil.propertyAsString(properties, DYNAMODB_TABLE_NAME,
         DYNAMODB_TABLE_NAME_DEFAULT);
+
+    this.s3WriteTags = toTags(properties, S3_WRITE_TAGS_PREFIX);
```

Comment:
we should also initialize the s3WriteTags value in the constructor above : 
https://github.com/apache/iceberg/pull/4350/files#diff-19af8af4e4204814f89c7e71633aae4b635229472dd20772d7651044d20cc900R291

```java
this.s3WriteTags = Sets.newHashSet();
```

otherwise the people using the constructor will get NPE.

Reply:
Ahh I see you made a check of tags != null below then it might not be required, but I think still empty set as default is good as it avoids us checking the null checks when it's being used. WDYT ?

Reply:
I agree, empty set would be a cleaner solution. I have made the changes.

Reply:
Nice catch for this! I think we still need to check null, because the object is serialized, and it might serialize an old version of the object that does not have the `writeTags` fields, which means the object will be deserialized with the value equal to null.

Also nit: similar to the above comments, could you move the assignment to L347?